kernel/src/transaction/mod.rs

```diff
@@ -297,16 +311,29 @@ impl WriteContext {
     }
 }
 
-/// Result after committing a transaction. If 'committed', the version is the new version written
-/// to the log. If 'conflict', the transaction is returned so the caller can resolve the conflict
-/// (along with the version which conflicted).
-// TODO(zach): in order to make the returning of a transaction useful, we need to add APIs to
-// update the transaction to a new version etc.
+/// Kernel exposes information about the state of the table that engines might want to use to
+/// trigger actions like checkpointing or log compaction. This struct holds that information.
+#[derive(Debug)]
+pub struct PostCommitStats {
+    pub commits_since_checkpoint: usize,
+    pub commits_since_log_compaction: usize,
```

Comment:
since these are effectively counters (unrelated to mem size etc.) should we just make them `u64` instead?

Reply:
makes sense, will update